pkg/container-utils/containerd/containerd.go

```diff
@@ -15,31 +15,185 @@
+		client: client,
+	}, nil
+}
+
+func (c *ContainerdClient) Close() error {
+	if c.client != nil {
+		return c.client.Close()
+	}
+	return nil
+}
+
+func (c *ContainerdClient) GetContainers() ([]*runtimeclient.ContainerData, error) {
+	taskResponse, err := c.client.TaskService().List(context.TODO(), &tasks.ListTasksRequest{})
+	if err != nil {
+		return nil, err
+	}
+
+	ret := make([]*runtimeclient.ContainerData, len(taskResponse.Tasks))
+	for i, task := range taskResponse.Tasks {
+		ret[i], err = c.ContainerdTaskToContainerData(task)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return ret, nil
+}
+
+func (c *ContainerdClient) GetContainer(containerID string) (*runtimeclient.ContainerData, error) {
+	response, err := c.client.TaskService().Get(context.TODO(), &tasks.GetRequest{
+		ContainerID: containerID,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	containerData, err := c.ContainerdTaskToContainerData(response.Process)
+	if err != nil {
+		return nil, err
+	}
+	return containerData, nil
+}
+
+func (c *ContainerdClient) GetContainerDetails(containerID string) (*runtimeclient.ContainerDetailsData, error) {
+	response, err := c.client.TaskService().Get(context.TODO(), &tasks.GetRequest{
+		ContainerID: containerID,
+	})
+	if err != nil {
+		return nil, err
+	}
+	proc := response.Process
+
+	containerData, err := c.ContainerdTaskToContainerData(proc)
+	if err != nil {
+		return nil, err
+	}
+
+	containers, err := c.client.Containers(context.TODO(), fmt.Sprintf("id==%s", containerID))
+	if err != nil {
+		return nil, err
+	}
+	if len(containers) != 1 {
+		return nil, fmt.Errorf("expected 1 container with id %q, got %d", containerID, len(containers))
+	}
+
+	spec, err := containers[0].Spec(context.TODO())
+	if err != nil {
+		return nil, err
+	}
+
+	mountData := make([]runtimeclient.ContainerMountData, len(spec.Mounts))
+	for i := range spec.Mounts {
+		mount := spec.Mounts[i]
+		mountData[i] = runtimeclient.ContainerMountData{
+			Source:      mount.Source,
+			Destination: mount.Destination,
+		}
+	}
+
+	return &runtimeclient.ContainerDetailsData{
+		ContainerData: *containerData,
+		Pid:           int(proc.Pid),
+		CgroupsPath:   spec.Linux.CgroupsPath,
+		Mounts:        mountData,
 	}, nil
 }
+
+// Convert the state from container status to state of runtime client.
+func processStatusStateToRuntimeClientState(status task.Status) (runtimeClientState string) {
+	switch status {
+	case task.StatusCreated:
+		runtimeClientState = runtimeclient.StateCreated
+	case task.StatusRunning:
+		runtimeClientState = runtimeclient.StateRunning
+	case task.StatusStopped:
+		runtimeClientState = runtimeclient.StateExited
+	default:
+		runtimeClientState = runtimeclient.StateUnknown
+	}
+	return
+}
+
+func guessContainerName(container containerd.Container) string {
+	labels, err := container.Labels(context.TODO())
+	if err != nil {
+		return container.ID()
+	}
+
+	if nerdctlName, ok := labels[ContainerNameNerdctl]; ok {
+		return nerdctlName
```

Comment:
What is the nerdctlName? When is it set?

Reply:
Directly in the `if` statement above.

Reply:
I mean, what does it represent? A different name from the Kubernetes container name?

Reply:
`nerdctl` saves the containername in that label when it creates a `containerd` container.
But if we currently only support K8s namespace it does not make that much sense.